frontend-runitup/src/Components/Community/ClubChat.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useEffect, useRef } from "react";
+import { Input, Button, Avatar, Typography, Spin, message } from "antd";
+import { SendOutlined } from "@ant-design/icons";
+import { getHeaders, generateColor } from "../../utils/apiConfig";
+
+const { Text } = Typography;
+
+const ClubChat = ({ clubId, currentUser }) => {
+  const [messages, setMessages] = useState([]);
+  const [newMessage, setNewMessage] = useState("");
+  const [loading, setLoading] = useState(true);
+  const messagesEndRef = useRef(null);
+  const chatContainerRef = useRef(null);
+
+  useEffect(() => {
+    fetchMessages();
+    const intervalId = setInterval(fetchMessages, 5000);
+    return () => clearInterval(intervalId);
+  }, [clubId]);
+
+  useEffect(() => {
+    scrollToBottom();
+  }, [messages]);
```

Comment:
nit: can't you move this logic within `fetchMessages` and remove the `useEffect` altogether? i.e. put `scrollToBottom` after `setMessages(data);` below within `fetchMessages`? Or is there some other behavior this is covering that I'm missing

Reply:
Addressed the nit and resolved it. You are correct we can remove the useEffect and do it in the fetch.